pairing/bn256/scalar.go

```diff
@@ -20,8 +21,15 @@ func newScalar() kyber.Scalar {
 }
 
 func (s *scalar) Equal(a kyber.Scalar) bool {
-	x := a.(*scalar).x
-	return subtle.ConstantTimeCompare(s.x.Bytes(), x.Bytes()) == 1
+	sm, err := s.MarshalBinary()
+	if err != nil {
+		return false
+	}
+	am, err := a.MarshalBinary()
+	if err != nil {
+		return false
+	}
+	return subtle.ConstantTimeCompare(sm, am) == 1
```

Comment:
not sure if this is crucial, but having `if err != nil {` makes this not constant time, no?

Reply:
I'm not even sure that `MarshalBinary` is constant-time. Most probably not, because it uses `Bytes()` which can return a variable number of bytes. So we might as well be honest and say it's variable time and use `bytes.Compare` ;)

Reply:
Yeah it's redundant. I removed those ifs and let the constant time comparison deal with errors from the marshaling.